### PR TITLE
ignore contents of None for write_file

### LINKED ISSUE #1066: 'NoneType' object has no attribute 'encode'
We have noticed failures with CustomScriptForLinux with the following error message.

```text
'NoneType' object has no attribute 'encode'
```

This may affect other extensions in this repo, but the only report we have is for CustomScript.

### PR DESCRIPTION
**Pull request by boumenot**
Closes #1066

### CODE REVIEW

File: azurelinuxagent/common/utils/fileutil.py [hunk 1]
```diff
@@ -76,6 +76,9 @@ def write_file(filepath, contents, asbin=False, encoding='utf-8', append=False):
     """
     Write 'contents' to 'filepath'.
     """
+    if contents is None:
```
Comment (jasonzio) [inline azurelinuxagent/common/utils/fileutil.py]: Are you _really_ sure this is what you want? If contents==None, then the file won't even be created. I would think that write_file(path, None) would create an empty file (file exists but is of zero length). That's the existing behavior, isn't it? I would expect this to break something.

File: tests/utils/test_file_util.py [hunk 2]
```diff
@@ -38,7 +38,16 @@ def test_read_write_file(self):
...
         self.assertEquals(content, content_read)
         os.remove(test_file)
-    
+
+    def test_write_empty_file(self):
+        """
+        write_file should ignore contents == None, and do nothing
+        """
+        test_file=os.path.join(self.tmp_dir, self.test_file)
+        fileutil.write_file(test_file, None)
```
Comment (jasonzio) [inline tests/utils/test_file_util.py]: If the file exists before write_file(path, None) is called, then your new behavior results in the file being untouched. The previous behavior for results in the file being truncated to zero length when append is False. You're changing behavior. If you desire this new behavior, you should have explicit tests that address the permutations.

File: azurelinuxagent/common/utils/fileutil.py (hunk 1 above)
Comment (jasonzio) [reply azurelinuxagent/common/utils/fileutil.py]: From our office conversation: the post-condition for this method should be "In the absence of a raised exception, a file named 'filepath' will exist on disk". I don't care if "contents==None => Exception" or "contents==None  => create empty file"; either one is acceptable.

File: tests/utils/test_file_util.py (hunk 2 above)
Comment (jasonzio) [reply tests/utils/test_file_util.py]: This test is badly named; you're not writing an empty file, you're writing None. `contents=""` is different from `contents=None`.

File: tests/ga/test_extension.py [hunk 3]
```diff
@@ -233,6 +233,24 @@ def test_migration_does_not_overwrite(self):
...
+
+        status = "Ready"
+        code = 0
+        message = "A message"
+
+        try:
+           with patch('json.dumps', return_value=None):
+                self.ext_handler_i.set_handler_status(status=status, code=code, message=message)
+        except Exception as e:
+            self.fail("set_handler_status threw and exception")
```
Comment (jasonzio) [inline tests/ga/test_extension.py]: typo (threw _an_ exception)

Comment (boumenot) [reply tests/ga/test_extension.py]: Fixed.

**Pull request merged**